File: Python/108. Convert Sorted Array to Binary Search Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sortedArrayToBST(self, nums: List[int]) -> Optional[TreeNode]:
        def sol(l, r):
            if l>r:
                return None

            m = (l+r)//2
            root= TreeNode(nums[m])
            root.left=sol(l,m-1)
            root.right=sol(m+1,r)
            return root
        return sol(0,len(nums)-1)

File: C#/108. Convert Sorted Array to Binary Search Tree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public TreeNode sol(int l, int r, int[] nums){
        if(l>r){
            return null;
        }
        int m = (l+r)/2;
        TreeNode root= new TreeNode(nums[m]);
        root.left=sol(l,m-1,nums);
        root.right=sol(m+1,r,nums);
        return root;
    }
    public TreeNode SortedArrayToBST(int[] nums) {
        return sol(0,nums.Length-1,nums);
    }
}
